cli/internal/cmd/apply_test.go

```diff
@@ -167,5 +177,215 @@ func TestSkipPhases(t *testing.T) {
+		createTfState      func(require *require.Assertions, fh file.Handler)
+		stdin              string
+		flags              applyFlags
+		wantPhases         skipPhases
+		wantErr            bool
+	}{
+		"gcp: all files exist": {
+			createConfig:       defaultConfig(cloudprovider.GCP),
+			createState:        defaultState,
+			createMasterSecret: defaultMasterSecret,
+			createAdminConfig:  defaultAdminConfig,
+			createTfState:      defaultTfState,
+			flags:              applyFlags{},
+			wantPhases: skipPhases{
+				skipInitPhase: struct{}{},
+			},
+		},
+		"aws: all files exist": {
+			createConfig:       defaultConfig(cloudprovider.AWS),
+			createState:        defaultState,
+			createMasterSecret: defaultMasterSecret,
+			createAdminConfig:  defaultAdminConfig,
+			createTfState:      defaultTfState,
+			flags:              applyFlags{},
+			wantPhases: skipPhases{
+				skipInitPhase: struct{}{},
+			},
+		},
+		"azure: all files exist": {
+			createConfig:       defaultConfig(cloudprovider.Azure),
+			createState:        defaultState,
+			createMasterSecret: defaultMasterSecret,
+			createAdminConfig:  defaultAdminConfig,
+			createTfState:      defaultTfState,
+			flags:              applyFlags{},
+			wantPhases: skipPhases{
+				skipInitPhase: struct{}{},
+			},
+		},
+		"qemu: all files exist": {
+			createConfig:       defaultConfig(cloudprovider.QEMU),
+			createState:        defaultState,
+			createMasterSecret: defaultMasterSecret,
+			createAdminConfig:  defaultAdminConfig,
+			createTfState:      defaultTfState,
+			flags:              applyFlags{},
+			wantPhases: skipPhases{
+				skipInitPhase:  struct{}{},
+				skipImagePhase: struct{}{}, // No image upgrades on QEMU
+			},
+		},
+		"no config file": {
+			createConfig:       func(require *require.Assertions, fh file.Handler) {},
+			createState:        defaultState,
+			createMasterSecret: defaultMasterSecret,
+			createAdminConfig:  defaultAdminConfig,
+			createTfState:      defaultTfState,
+			flags:              applyFlags{},
+			wantErr:            true,
+		},
+		"no admin config file, but mastersecret file exists": {
+			createConfig:       defaultConfig(cloudprovider.GCP),
+			createState:        defaultState,
+			createMasterSecret: defaultMasterSecret,
+			createAdminConfig:  func(require *require.Assertions, fh file.Handler) {},
+			createTfState:      defaultTfState,
+			flags:              applyFlags{},
+			wantErr:            true,
+		},
+		"no admin config file, no master secret": {
+			createConfig:       defaultConfig(cloudprovider.GCP),
+			createState:        defaultState,
+			createMasterSecret: func(require *require.Assertions, fh file.Handler) {},
+			createAdminConfig:  func(require *require.Assertions, fh file.Handler) {},
+			createTfState:      defaultTfState,
+			flags:              applyFlags{},
+			wantPhases: skipPhases{
+				skipImagePhase: struct{}{},
+				skipK8sPhase:   struct{}{},
+			},
+		},
+		"no tf state, but admin config exists": {
```

Comment:
this should not error, for self-managed so let's add this:
```suggestion
		"error if not skip infra and no tf state, but admin config exists": {
```

Reply:
add a test case for self managed

Reply:
> this should not error, for self-managed

For self-managed skipInfrastructure has to be set, otherwise it should error (which this test case checks)

> add a test case for self managed

See the last test case of this function

Reply:
looks much better thanks! can you add a case for self-managed upgrade